### PR TITLE
[NL] Display fixes to overviews with map only

### PR DESCRIPTION
**Pull request by juliawu**
Fixes the emtpy whitespace on overview tiles in the NL interface when no ranking data is available.

## Before:
Screenshot 2023-02-17 at 2 04 25 PM

## After:
Screenshot 2023-02-17 at 2 04 39 PM

## Validations
Also verified changes don't break other NL queries, the place pages, or demo scripts:

Screenshot 2023-02-17 at 2 05 29 PM
Screenshot 2023-02-17 at 2 05 59 PM
Screenshot 2023-02-17 at 2 06 13 PM
Screenshot 2023-02-17 at 2 06 28 PM
(no map expected for USA)

### CODE REVIEW

File: static/js/place/overview.tsx [hunk 1]
```diff
@@ -30,24 +32,66 @@ interface OverviewPropType {
...
-              locale={this.props.locale}
-            ></Ranking>
+          <div
+            className={`col-12 ${this.shouldShowRanking() ? "col-md-4" : ""}`}
+          >
+            <div className={this.shouldShowRanking() ? "map-with-margin" : ""}>
+              <GoogleMap dcid={this.props.dcid}></GoogleMap>
+            </div>
           </div>
+          {this.shouldShowRanking() ? (
```
Comment (chejennifer) [inline static/js/place/overview.tsx:63]: can make this check once at the top of the render function & save it in a constant : `const shouldShowRanking = this.shouldShowRanking()`

File: static/js/place/overview.tsx [hunk 2]
```diff
@@ -30,24 +32,66 @@ interface OverviewPropType {
...
+      .then((resp) => {
+        this.setState({ rankingData: resp.data });
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }
+
+  shouldShowRanking(): boolean {
+    return !_.isEmpty(this.state.rankingData.label);
```
Comment (chejennifer) [inline static/js/place/overview.tsx:93]: nit: this doesn't need its own function

File: static/js/place/overview.tsx [hunk 3]
```diff
@@ -30,24 +32,66 @@ interface OverviewPropType {
...
+              <div className="col-12 col-md-8">
+                <Ranking
+                  dcid={this.props.dcid}
+                  locale={this.props.locale}
+                  data={this.state.rankingData}
+                ></Ranking>
+              </div>
+            </>
+          ) : (
+            <></>
```
Comment (chejennifer) [inline static/js/place/overview.tsx:74]: this doesn't need to be an if else check, can just do `shouldShowRanking && <div ...><Ranking .../></div> ` which says only render the ranking div if shouldShowRanking

File: static/css/nl_interface.scss [hunk 4]
```diff
@@ -193,9 +193,13 @@ $answer-min-height: 100vh;
       padding: 0.25rem;
     }
 
+    .map-with-margin {
+      margin-right: 15px;
+      height: 100%;
```
Comment (chejennifer) [inline static/css/nl_interface.scss:198]: do we need this `height:100%?` isn't height always going to be 100% with or without the ranking section

File: static/js/place/overview.tsx [hunk 5]
```diff
@@ -30,24 +32,66 @@ interface OverviewPropType {
...
-          </div>
-          <div className="col-12 col-md-8">
-            <Ranking
-              dcid={this.props.dcid}
-              locale={this.props.locale}
-            ></Ranking>
+          <div
+            className={`col-12 ${this.shouldShowRanking() ? "col-md-4" : ""}`}
+          >
+            <div className={this.shouldShowRanking() ? "map-with-margin" : ""}>
```
Comment (chejennifer) [inline static/js/place/overview.tsx:59]: do we need this extra div? can we just do `col-12 ${this.shouldShowRanking() ? "col-md-4 map-with-margin" : ""}` in line 57

**Review by chejennifer — COMMENTED** [review]
thanks!

File: static/js/place/overview.tsx (hunk 5 above)
Comment (juliawu) [reply static/js/place/overview.tsx:59]: When I tried to combine map-with-margin in the same div as col-md-4, the rankings would show up under the map instead
image

File: static/css/nl_interface.scss (hunk 4 above)
Comment (juliawu) [reply static/css/nl_interface.scss:198]: When height isn't set here, the maps on the place pages don't take the full height (instead, they're stuck at 160px min height).

File: static/js/place/overview.tsx (hunk 3 above)
Comment (juliawu) [reply static/js/place/overview.tsx]: good catch! Switched to using &&

File: static/js/place/overview.tsx (hunk 1 above)
Comment (juliawu) [reply static/js/place/overview.tsx]: Ah, this is a much better idea than the helper function I tried to use instead! Switched to saving a constant at the top.

File: static/js/place/overview.tsx (hunk 2 above)
Comment (juliawu) [reply static/js/place/overview.tsx]: removed the function

File: static/js/place/overview.tsx (hunk 5 above)
Comment (juliawu) [reply static/js/place/overview.tsx]: Played around with the CSS more and was able to remove the extra div!

File: static/css/nl_interface.scss (hunk 4 above)
Comment (juliawu) [reply static/css/nl_interface.scss]: Played around with this more: now that I removed the extra div, height is no longer set here.

File: static/css/place/place.scss [hunk 6]
```diff
@@ -102,6 +102,10 @@ h3 {
   text-decoration: underline;
 }
 
+.map-with-margin {
+  height: 100%;
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/css/place/place.scss]: CSS has been cleaned up (and renamed to map-with-padding because padding is now used instead.

**Comment by juliawu** [comment]
Some updated screenshots after CSS changes. I saw no changes to place pages in my testing:

Screenshot 2023-02-17 at 7 01 45 PM
Screenshot 2023-02-17 at 7 02 00 PM

**Comment by beets** [comment]
@pradh - fyi: this also updates "Learn more about" to "Learn about" due to the way components are laid out. we can merge if this becomes an issue (or hold off to avoid code changes). otherwise should be ready for merging.

**Comment by juliawu** [comment]
These changes superseded by PR #2281

**Pull request closed without merging**